packages/flutter/test/material/text_button_test.dart

```diff
@@ -1009,6 +1009,72 @@ void main() {
     expect(paddingWidget.padding, const EdgeInsets.all(22));
   });
 
+  testWidgets('TextButton.icon is configurable when iconAlignment set to ButtonIconAlignment.start', (WidgetTester tester) async {
+    const Widget icon = Icon(Icons.add);
+    const ButtonIconAlignment buttonIconAlignment = ButtonIconAlignment.start;
+    const Widget label = Text('text button');
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: TextButton.icon(
+            onPressed: () { },
+            icon: icon,
+            iconAlignment: buttonIconAlignment,
+            label: label,
+          ),
+        ),
+      ),
+    );
+
+    final Finder textButtonWithIconFinder = find.byWidgetPredicate((Widget widget) => '${widget.runtimeType}' == '_TextButtonWithIconChild');
+
+    final dynamic textButtonWithIconWidget = tester.widget(textButtonWithIconFinder);
+    expect(textButtonWithIconWidget.iconAlignment, buttonIconAlignment);
+
+    final Row rowInsideTextButton = tester.widget(
+      find.descendant(
+        of: textButtonWithIconFinder,
+        matching: find.byType(Row),
+      ),
+    );
+    expect(icon, rowInsideTextButton.children.first);
+    expect(label, rowInsideTextButton.children.last);
+  });
+
+  testWidgets('TextButton.icon is configurable when iconAlignment set to ButtonIconAlignment.end', (WidgetTester tester) async {
+    const Widget icon = Icon(Icons.add);
+    const ButtonIconAlignment buttonIconAlignment = ButtonIconAlignment.end;
+    const Widget label = Text('text button');
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: TextButton.icon(
+            onPressed: () { },
+            icon: icon,
+            iconAlignment: buttonIconAlignment,
+            label: label,
+          ),
+        ),
+      ),
+    );
+
+    final Finder textButtonWithIconFinder = find.byWidgetPredicate((Widget widget) => '${widget.runtimeType}' == '_TextButtonWithIconChild');
+
+    final dynamic textButtonWithIconWidget = tester.widget(textButtonWithIconFinder);
+    expect(textButtonWithIconWidget.iconAlignment, buttonIconAlignment);
+
+    final Row rowInsideTextButton = tester.widget(
+      find.descendant(
+        of: textButtonWithIconFinder,
+        matching: find.byType(Row),
+      ),
+    );
+    expect(label, rowInsideTextButton.children.first);
+    expect(icon, rowInsideTextButton.children.last);
+  });
+
 }
```

Comment:
we should also have a test that verifies that the alignment works when you change the text direction

Reply:
text direction will flip it

Reply:
For example, check the position of the icon with one direction, then check it with the opposite direction.

Reply:
ah okay

Reply:
I mean the `textDirection` flips the concept if you look at it from a `ltr` language, but for example, if you look at it when text direction is 'rtl' it makes sense that start is at the right because that's where people start reading from in `rtl` languages. I think it is not needed but it is your call ✌. The point of using `ButtonIconAlignment` is to put the icon at the end or at the start complying with the text direction. I really don't get it why we need a test here? 😉🙌